Assignment 3

kevinlutoronto opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

Tried completing Assignment 3.

## What did you learn from the changes you have made?

Learned about bootstrap, confidence intervals and K-Means.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?

None. Just complete the assignment.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?

None.

## How were these changes tested?

Yes.

## A reference to a related issue in your repository (if applicable)

None

## Checklist
- [x] I can confirm that my changes are working as intended

kevinlutoronto2026 committed:
1st commit of assignment_1.ipynb, all questions answered.

kevinlutoronto2026 committed:
2nd commit, added more answer to Question 2 ii

kevinlutoronto2026 committed:
1st commit of LCR assignment-2

kevinlutoronto2026 committed:
2nd commit of assignment_2.ipynb

kevinlutoronto2026 committed:
3rd commit of assignment_2.ipynb

kevinlutoronto2026 committed:
1st commit of assignment_3.ipynb

kevinlutoronto2026 committed:
2nd commit of assignment-3.ipynb

kevinlutoronto2026 committed:
3rd commit of assignment_3.ipynb

PatelVishakh reviewed (changes requested):
Assignment 3 incomplete.

Comments:

Q2) detecting if clusters occur in the our data can be used as an indicator of different species which is the goal of the question.

Q5) Code) you are creating new data, you should be using the iris data set.

kevinlutoronto2026 committed:
4th commit of assignment_3.ipynb, this commit is already receiving email saying this assignment is not completed

kevinlutoronto2026 committed:
5th commit of assignment_3.ipynb, this commit is already receiving email saying this assignment is not completed

kevinlutoronto commented:
> Assignment 3 incomplete.
> 
> Comments:
> 
> Q2) detecting if clusters occur in the our data can be used as an indicator of different species which is the goal of the question.
> 
> Q5) Code) you are creating new data, you should be using the iris data set.

Hi Vishakh,

Thank you for your feedback. I have updated the answers to Questions 2 and 5. Please take a look. The committed's name is "5th commit of assignment_3.ipynb, this commit is already receiving em…"

Thank you!

Kevin

PatelVishakh reviewed (changes requested):
Assignment 3 Incomplete:

Please make a new pull request (instead of a commit) with the changes requested.

Q2) Linear regression is not suited for this setting because we are trying to detect different species in our data, and for this approach clustering is the correct approach, while linear regression is there to model the relationship between an independent variable and dependent var. 

Q5) Code output is incorrect. One of the issues is  need to change current_sample to using the full data in "for i in range(0, 20000):
    s = current_sample.sample(frac = 1, replace = True).assign(sample_id = i)
    sample_list.append(s)".

kevinlutoronto2026 committed:
1st commit after assignment_3 is deemed incomplete for 2nd time

kevinlutoronto commented:
> Assignment 3 Incomplete:
> 
> Please make a new pull request (instead of a commit) with the changes requested.
> 
> Q2) Linear regression is not suited for this setting because we are trying to detect different species in our data, and for this approach clustering is the correct approach, while linear regression is there to model the relationship between an independent variable and dependent var.
> 
> Q5) Code output is incorrect. One of the issues is need to change current_sample to using the full data in "for i in range(0, 20000): s = current_sample.sample(frac = 1, replace = True).assign(sample_id = i) sample_list.append(s)".

Hi Vishakh, sorry I didn't know how to create a new pull request as it the terminal seems not to be able to do that. So I commit a new assignment_3.ipynb under the name "1st commit after assignment_3 is deemed incomplete for 2nd time." I modified the code stated in your feedback to 10000 in Question 5 and modified the Question 2 explaining why Clustering is better than Linear Regression for this question. Thanks!

PatelVishakh reviewed (approved):
Assignment 3: complete, nice work.

Final Comments

Please fix the PR issue when you have a chance.

Also in Q5) Code) replace all "Current_sample" with "iris_df['petal width (cm)']"
